### DISCUSSION TITLE
Channel Closing

### DISCUSSION THREAD

**Discussion by gustavohfelixs** [rabbitmq/rabbitmq-java-client#1074] [General]
### Describe the bug

Hello Guys.
I'm using Apache Camel and RabbitMq on a project to make an update in some itens status, if the item doesn't exists in database i need to purge the message. After pass and return by Router of Apache when `channel.basicAck() ` is called throws an exception,. Follow the snippets

```
 public void readMessage(@Payload final String payload, Channel channel, @Header(AmqpHeaders.DELIVERY_TAG) long tag
    ) throws IOException {
        try {
            final var myObject = objectMapper.readValue(payload, myObject.class);

            log.info("Object received : {}", myObject );

            service.paymentObject(myObject);

            channel.basicAck(tag, false);
```

Exception: `org.springframework.amqp.AmqpApplicationContextClosedException: The ApplicationContext is closed and the ConnectionFactory can no longer create connections.`

How can I fix that? 

Specs:
Spring: 2.2.0
JDK 11 - Oracle OpenJDK version 11.0.6
Camel: 3.0.1

### Reproduction steps

1. Receive the message in Listener
2. Go to Router (Apache Camel), try to verify if value exists in other components
3. If the value doens't exists purge the message of queue (Now its throwing exception)
...

### Expected behavior

To Handle the exception and purge the invalid messages of queue

**Comment by michaelklishin** [comment]
We cannot suggest much without [server logs](https://rabbitmq.com/logging.html#log-file-location). I'm afraid our team won't be setting up Apache Camel to reproduce and troubleshoot this. You can put together an executable example in a repo but logs and a traffic will likely be enough.

My best guess right now is that you try to ack on a channel that's different from the channel on which you have received the delivery. That results in a [channel exception](https://rabbitmq.com/channels.html#error-handling). Whether that can result in an "application context closure" in Spring AMQP, I don't know, but doing so will result in a channel exception that renders the channel unusable.

Double acking or acking when the consumer was set up to use [automatic acknowledgement](https://rabbitmq.com/confirms.html#acknowledgement-modes) would have the same effect.
